src/Containers/Impl/Docker/LocalDockerExtensions.cs

```diff
@@ -0,0 +1,98 @@
+
+        public static Task<string> RepositoryLogoutAsync(this LocalDocker dockerInfo, IProcessServices ps, RepositoryCredentials auth, CancellationToken ct) {
+            return dockerInfo.RepositoryLogoutAsync(ps, auth.RepositoryServer, ct);
+        }
+
+        public static async Task<string> RepositoryLogoutAsync(this LocalDocker dockerInfo, IProcessServices ps, string server, CancellationToken ct) {
+            string command = "logout";
+            string commandOptions = server;
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {commandOptions}", -1, ct);
+        }
+
+        public static async Task<string> PullImageAsync(this LocalDocker dockerInfo, IProcessServices ps, string fullImageName, CancellationToken ct) {
+            string command = "pull";
+            string commandOptions = fullImageName;
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {commandOptions}", -1, ct);
+        }
+
+        public static async Task<string> CreateContainerAsync(this LocalDocker dockerInfo, IProcessServices ps, string createOptions, CancellationToken ct) {
+            string command = "create";
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {createOptions}", -1, ct);
+        }
+
+        public static async Task<string> DeleteContainerAsync(this LocalDocker dockerInfo, IProcessServices ps, IContainer container, CancellationToken ct) {
+            string command = "rm";
+            string commandOptions = $"{container.Id}";
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {commandOptions}", -1, ct);
+        }
+
+        public static async Task<string> StartContainerAsync(this LocalDocker dockerInfo, IProcessServices ps, IContainer container, CancellationToken ct) {
+            string command = "start";
+            string commandOptions = $"{container.Id}";
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {commandOptions}", -1, ct);
+        }
+
+        public static async Task<string> StopContainerAsync(this LocalDocker dockerInfo, IProcessServices ps, IContainer container, CancellationToken ct) {
+            string command = "stop";
+            string commandOptions = $"{container.Id}";
+            return await ExecuteCommandAsync(ps, dockerInfo.Docker, $"{command} {commandOptions}", -1, ct);
+        }
+
+        private static async Task<string> ExecuteCommandAsync(IProcessServices ps, string commandFile, string arguments, int timeoutms = 5000, CancellationToken ct = default(CancellationToken)) {
+            ProcessStartInfo psi = new ProcessStartInfo();
+            psi.FileName = commandFile;
+            psi.Arguments = arguments;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardInput = true;
+            psi.RedirectStandardOutput = true;
+            psi.UseShellExecute = false;
+
+            var process = ps.Start(psi);
```

Comment:
The problem is that this depends on the state of the docker cache. If it is a fresh install, and the connection is slow. This can take a lot of time (say ~20-30 mins), since it will have to download ubuntu image + packages.